pkg/apiserver/resthandler.go

```diff
@@ -347,6 +347,21 @@ func DeleteResource(r RESTDeleter, ctxFn ContextFunc, namer ScopeNamer, codec ru
 			ctx = api.WithNamespace(ctx, namespace)
 		}
 
+		options := &api.DeleteOptions{}
+		if checkBody {
+			body, err := readBody(req.Request)
+			if err != nil {
+				errorJSON(err, codec, w)
+				return
+			}
+			if len(body) > 0 {
```

Comment:
I'd like to preserve the nil behavior and then potentially remove support in the future.  It's not out of place in our codebase for nil to mean something different than passing an initialized DeleteOptions(0), and once we have a few implementations graceful delete I'll feel more comfortable about removing the old callers of Delete(2)